### PR TITLE
[Grade] - Use "versions.gradle" to organize dependencies

### PR DESCRIPTION
**Pull request by jaredsburrows**
@vanniktech

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
-        classpath 'com.vanniktech:gradle-android-junit-jacoco-plugin:0.4.0'
+        classpath plugs.kotlinGradlePlugin
+        classpath plugs.pluginPublishPlugin
+        classpath plugs.gradleVersionsPlugin
+        classpath plugs.gradleAndroidJunitJacocoPlugin
     }
 }
 
 apply plugin: 'groovy'
+apply plugin: 'java-gradle-plugin'
```
Comment (jaredsburrows) [inline build.gradle:17]: Use new Java library plugin.

File: gradle/versions.gradle [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  sourceCompatibilityVersion = JavaVersion.VERSION_1_7
+  targetCompatibilityVersion = JavaVersion.VERSION_1_7
+
+  // Common versions
+  kotlinVersion = "1.1.3-2"
+}
+
+// Plugins
+ext.plugs = [
+        "gradle"                        : "com.android.tools.build:gradle:2.3.3",
```
Comment (vanniktech) [inline gradle/versions.gradle]: can you also do 2 spaces here?

File: build.gradle [hunk 3]
```diff
@@ -1,17 +1,20 @@
 buildscript {
+    apply from: rootProject.file("gradle/versions.gradle")
```
Comment (vanniktech) [inline build.gradle]: can we call it dependencies.gradle and move it into the root directory?

File: gradle/versions.gradle [hunk 4]
```diff
@@ -0,0 +1,29 @@
+// Variables for entire project
+ext {
+  // Setup
+  sourceCompatibilityVersion = JavaVersion.VERSION_1_7
+  targetCompatibilityVersion = JavaVersion.VERSION_1_7
```
Comment (vanniktech) [inline gradle/versions.gradle]: this one is unused, right?

Comment (jaredsburrows) [reply gradle/versions.gradle]: I'll remove it.

File: gradle/versions.gradle (hunk 2 above)
Comment (jaredsburrows) [reply gradle/versions.gradle]: Fixed and I'll add an ".editorconfig" to help all new comers to this project.

File: build.gradle (hunk 3 above)
Comment (jaredsburrows) [reply build.gradle]: I renamed it to "dependencies.gradle" but why not keep it in the "gradle" folder? You have "gradle-mvn-push.gradle" in there as well.

Comment (vanniktech) [reply build.gradle]: I've got the file in all my other projects in the root directory.

File: gradle/versions.gradle (hunk 2 above)
Comment (vanniktech) [reply gradle/versions.gradle]: 👍

**Comment by jaredsburrows** [comment]
@vanniktech Updated. Let me know what you think.

File: .editorconfig [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+indent_style = space
+indent_size = 2
+
+# We recommend you to keep these unchanged
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+# Markdown files sometimes need trailing whitespaces.
```
Comment (vanniktech) [inline .editorconfig]: how come? I've never had to keep it

Comment (jaredsburrows) [reply .editorconfig]: I just added this to automatically enforce the 2 space indentation that you wanted.

Comment (vanniktech) [reply .editorconfig]: Okay can you set the trailnig whitespace to true though?

Comment (jaredsburrows) [reply .editorconfig]: Its from the example on the website: http://editorconfig.org/.

Comment (vanniktech) [reply .editorconfig]: Alright but we don't need trailing whitespaces in the README.

Comment (jaredsburrows) [reply .editorconfig]: Removed. Anything else?

**Review by vanniktech — APPROVED** [review]

**Pull request merged**